slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+    def __init__(self, baseten_url: str, baseten_api_key: str) -> None:
+        self._baseten_url = baseten_url
+        self._baseten_env = _infer_env(baseten_url)
+        self._baseten_api_key = baseten_api_key
+        self._remote_provider: truss_remote.TrussRemote = self._create_remote_provider()
+
+    def deploy_truss(self, truss_root: Path) -> definitions.BasetenRemoteDescriptor:
+        tr = truss.load(str(truss_root))
+        model_name = tr.spec.config.model_name
+        assert model_name is not None
+
+        logging.info(f"Deploying model `{model_name}`.")
+        service = self._remote_provider.push(
+            tr, model_name=model_name, trusted=True, publish=False
+        )
+        if service is None:
+            raise ValueError()
+        service = cast(b10_service.BasetenService, service)
+
+        model_service = definitions.BasetenRemoteDescriptor(
+            b10_model_id=service.model_id,
+            b10_model_version_id=service.model_version_id,
+            b10_model_name=model_name,
+            b10_model_url=_model_url(self._baseten_env, service.model_id),
+        )
+        return model_service
+
+    def get_model(self, model_name: str) -> definitions.BasetenRemoteDescriptor:
+        query_string = f"""
+        {{
+        model_version(name: "{model_name}") {{
+            oracle{{
+                id
+                name
+                versions{{
+                    id
+                    current_deployment_status
+                }}
+            }}
+        }}
+        }}
+        """
+        try:
+            resp = self._post_graphql_query(query_string, retries=True)["data"][
+                "model_version"
+            ]["oracle"]
+        except Exception as e:
+            raise definitions.MissingDependencyError("Model cout not be found.") from e
+
+        model_id = resp["id"]
+        model_version_id = resp["versions"][0]["id"]
+        return definitions.BasetenRemoteDescriptor(
+            b10_model_id=model_id,
+            b10_model_version_id=model_version_id,
+            b10_model_url=_model_url(self._baseten_env, model_id),
+            b10_model_name=model_name,
+        )
+
+    def _create_remote_provider(self) -> truss_remote.TrussRemote:
+        remote_config = truss_remote.RemoteConfig(
+            name="baseten",
+            configs={
+                "remote_provider": "baseten",
+                "api_key": self._baseten_api_key,
+                "remote_url": self._baseten_url,
+            },
+        )
+        remote_factory.RemoteFactory.update_remote_config(remote_config)
+        return remote_factory.RemoteFactory.create(remote="baseten")
+
+    def _wait_for_model_to_be_ready(self, model_version_id: str) -> None:
+        logging.info(f"Waiting for model {model_version_id} to be ready")
+
+        def is_model_ready() -> ConditionStatus:
+            query_string = f"""
+            {{
+                model_version(id: "{model_version_id}") {{
```

Comment:
could we use the rest api for this: https://docs.baseten.co/api-reference/gets-a-models-deployment-by-id

Reply:
Yes, there's already a TODO. It's just that I know this works as is and changing it now requires a whole test/deployment loop. I'll do it soon =)